src/main/kotlin/no/nav/syfo/client/DiskresjonskodeService.kt

```diff
@@ -0,0 +1,18 @@
+package no.nav.syfo.client
+
+import com.ctc.wstx.exc.WstxException
+import java.io.IOException
+import no.nav.syfo.helpers.retry
+import no.nav.tjeneste.pip.diskresjonskode.DiskresjonskodePortType
+import no.nav.tjeneste.pip.diskresjonskode.meldinger.WSHentDiskresjonskodeRequest
+
+class DiskresjonskodeService(private val diskresjonskodePortType: DiskresjonskodePortType) {
+
+    suspend fun hentDiskresjonskode(ident: String): String = retry(
+            callName = "hent_diskresjonskode",
+            retryIntervals = arrayOf(500L, 1000L, 3000L, 5000L, 10000L, 60000L),
```

Comment:
Hvis vi prøver på nytt så mange ganger og venter så lenge mellom hver gang har det vel timet ut hos klienten som kalte api'et? 

Og, jeg er litt usikker på om en eventuell feilmelding vil være en av de to som er spesifisert som legal exceptions. Mener å huske at det ikke er det...

Reply:
Ja enig at vi burde justere tiden og kanskje litt færre retries?